julia/core.py

```diff
@@ -286,7 +289,11 @@ def __init__(self, init_julia=True, jl_init_path=None):
                     "Install PyCall by running the following line:\n"
                     """\tjulia -e 'Pkg.add("PyCall"); Pkg.update()'\n""")
             try:
-                self.call('pyinitialize(C_NULL)')
+                if os.name == "nt":
+                    # on windows C_NULL doesn't mean anything
+                    self.call('pyinitialize()')
```

Comment:
I think the problem is that `jl_load_dynamic_library(C_NULL)` on Windows is returning a reference to the running `libjulia` library (not what we want), whereas on Unix it returns `dlopen(NULL)` which is the handle for the main program (which is what we want).